From ef3fe32759439ac671a6cf122667fdac609bd578 Mon Sep 17 00:00:00 2001
From: Stephane Viau <sviau@codeaurora.org>
Date: Tue, 7 Jul 2015 16:42:07 -0400
Subject: [PATCH 31/41] ARM: dts: qcom: Enable MDSS for msm8996 on MTP board

Adding KMS connector nodes for msm8996 MTP board and enable DRM on
this board.

Change-Id: I8aaa34c295b2da903199bfb4c88212db0d0bd99d
Signed-off-by: Stephane Viau <sviau@codeaurora.org>
---
 arch/arm/boot/dts/qcom/msm8996-dtp.dtsi |  2 +
 arch/arm/boot/dts/qcom/msm8996-mtp.dtsi |  6 ++-
 arch/arm/boot/dts/qcom/msm8996-v2.dtsi  |  4 ++
 arch/arm/boot/dts/qcom/msm8996.dtsi     | 87 ++++++++++++++++++++++++++++++++-
 4 files changed, 97 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/qcom/msm8996-dtp.dtsi b/arch/arm/boot/dts/qcom/msm8996-dtp.dtsi
index 19d0527..1740922 100644
--- a/arch/arm/boot/dts/qcom/msm8996-dtp.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996-dtp.dtsi
@@ -459,9 +459,11 @@
 	qcom,mdss-pref-prim-intf = "dsi";
 };
 
+/*
 &mdss_fb2 {
 	status = "disabled";
 };
+*/
 
 &mdss_hdmi_tx {
 	status = "disabled";
diff --git a/arch/arm/boot/dts/qcom/msm8996-mtp.dtsi b/arch/arm/boot/dts/qcom/msm8996-mtp.dtsi
index dfd4162..ecb6a8c 100644
--- a/arch/arm/boot/dts/qcom/msm8996-mtp.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996-mtp.dtsi
@@ -338,7 +338,11 @@
 #include "msm8996-mdss-panels.dtsi"
 
 &mdss_mdp {
-	qcom,mdss-pref-prim-intf = "dsi";
+	connectors = <&mdss_hdmi>;
+};
+
+&mdss_hdmi {
+	status = "ok";
 };
 
 &mdss_dsi {
diff --git a/arch/arm/boot/dts/qcom/msm8996-v2.dtsi b/arch/arm/boot/dts/qcom/msm8996-v2.dtsi
index ea73899..621501c 100644
--- a/arch/arm/boot/dts/qcom/msm8996-v2.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996-v2.dtsi
@@ -480,6 +480,10 @@
 	gdsc-venus-supply = <&gdsc_venus>;
 };
 
+&mdss_hdmi {
+	hpd-gdsc-venus-supply = <&gdsc_venus>;
+};
+
 &mdss_dsi {
 	gdsc-venus-supply = <&gdsc_venus>;
 	qcom,core-supply-entries {
diff --git a/arch/arm/boot/dts/qcom/msm8996.dtsi b/arch/arm/boot/dts/qcom/msm8996.dtsi
index 5844fe3..b8989b77 100644
--- a/arch/arm/boot/dts/qcom/msm8996.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996.dtsi
@@ -219,7 +219,6 @@
 };
 
 #include "msm8996-ion.dtsi"
-#include "msm8996-mdss.dtsi"
 #include "msm8996-mdss-pll.dtsi"
 #include "msm8996-smp2p.dtsi"
 #include "msm8996-ipcrouter.dtsi"
@@ -3748,6 +3747,92 @@
 		clock-names = "gpu_clk";
 		clocks = <&clock_gpu clk_gpu_gx_gfx3d_clk>;
 	};
+
+	mdss_mdp: qcom,mdss_mdp@900000 {
+		compatible = "qcom,mdss_mdp";
+		reg = <0x00900000 0x90000>,
+		      <0x009b0000 0x1040>,
+		      <0x009b8000 0x1040>;
+		reg-names = "mdp_phys",
+			"vbif_phys",
+			"vbif_nrt_phys";
+		clocks = <&clock_mmss clk_mdss_ahb_clk>,
+			 <&clock_mmss clk_mdss_axi_clk>,
+			 <&clock_mmss clk_mdp_clk_src>,
+			 <&clock_mmss clk_mdss_mdp_vote_clk>,
+			 <&clock_mmss clk_smmu_mdp_axi_clk>,
+			 <&clock_mmss clk_mmagic_mdss_axi_clk>,
+			 <&clock_mmss clk_mdss_vsync_clk>;
+		clock-names = "iface_clk",
+			"bus_clk",
+			"core_clk_src",
+			"core_clk",
+			"iommu_clk",
+			"mmagic_clk",
+			"vsync_clk";
+		mmagic-supply = <&gdsc_mmagic_mdss>;
+		vdd-supply = <&gdsc_mdss>;
+		interrupt-parent = <&intc>;
+		interrupts = <0 83 0>;
+		interrupt-controller;
+		#interrupt-cells = <1>;
+		iommus = <&mdp_smmu 0>;
+	};
+
+	mdss_dsi: qcom,mdss_dsi@0 {
+		// TODO
+	};
+
+	mdss_dsi0: qcom,mdss_dsi_ctrl0@994000 {
+		// TODO
+	};
+
+	mdss_dsi1: qcom,mdss_dsi_ctrl1@996000 {
+		// TODO
+	};
+
+	mdss_hdmi_tx: qcom,hdmi_tx@0 {
+		// XXX Unused by KMS driver
+	};
+
+	mdss_hdmi: qcom,hdmi_tx@9a0000 {
+		compatible = "qcom,hdmi-tx-8996";
+		status = "disabled";
+		reg =	<0x009a0000 0x50c>,
+			<0x00070000 0x6158>,
+			<0x009e0000 0xfff>;
+		reg-names = "core_physical",
+			"qfprom_physical",
+			"hdcp_physical";
+		clocks = <&clock_mmss clk_mdss_mdp_vote_clk>,
+			 <&clock_mmss clk_mdss_ahb_clk>,
+			 <&clock_mmss clk_mdss_hdmi_clk>,
+			 <&clock_mmss clk_mdss_hdmi_ahb_clk>,
+			 <&clock_mmss clk_mdss_extpclk_clk>;
+		clock-names =
+			"mdp_core_clk",
+			"iface_clk",
+			"core_clk",
+			"alt_iface_clk",
+			"extp_clk";
+		interrupt-parent = <&mdss_mdp>;
+		interrupts = <8 0>;
+		hpd-gdsc-supply = <&gdsc_mdss>;
+		//core-vdda-supply = <&pm8994_l12>;
+		//core-vcc-supply = <&pm8994_s4>;
+		qcom,hdmi-tx-hpd-gpio = <&pm8994_mpps 4 0>;
+		pinctrl-names = "default", "sleep";
+		pinctrl-0 = <&mdss_hdmi_hpd_active
+			     &mdss_hdmi_ddc_active
+			     &mdss_hdmi_cec_active>;
+		pinctrl-1 = <&mdss_hdmi_hpd_suspend
+			     &mdss_hdmi_ddc_suspend
+			     &mdss_hdmi_cec_suspend>;
+
+		hdmi_audio: qcom,msm-hdmi-audio-rx {
+			compatible = "qcom,msm-hdmi-audio-codec-rx";
+		};
+	};
 };
 
 &gdsc_venus {
-- 
2.8.2

